src/data/proxy/DDBPreferenceCorrelationGraph.java

```diff
@@ -108,4 +115,18 @@ public Preference getPreference(String id, PreferenceCategory category) {
             return new DDBPreferenceAdapter(item).toObject();
         }
     }
+    
+    private List<Preference> getPreferences(Page<Item, ScanOutcome> page) {
+    	Iterator<Item> iterator = page.iterator();
+    	return Stream.generate(iterator::next).map(item -> new DDBPreferenceAdapter(item).toObject()).collect(Collectors.toList());
+    }
+
+	@Override
+	public Stream<List<Preference>> batchGetPreferences(
```

Comment:
I hadn't thought of an Iterator, but I think that would be the best idea. That way, it's very isolated from the underlying batch implementation. If you did that, you could even change some other signatures to produce Iterators and make that our new Collection<Preference> return type for this interface.